app/services/experimentation/data_export.rb

```diff
@@ -0,0 +1,195 @@
+module Experimentation
+  class DataExport
+    require "csv"
+
+    FOLLOWUP_CUTOFF = 10.days
+
+    attr_reader :experiment, :max_communications, :max_appointments, :max_past_visits, :notification_start_date, :aggregate, :cutoff_date
+
+    def initialize(name)
+      @experiment = Experimentation::Experiment.find_by!(name: name)
+      remind_ons = experiment.reminder_templates.pluck(:remind_on_in_days)
+      @notification_start_date = experiment.start_date - remind_ons.min.days
```

Comment:
Using this date to catch encounters that occur before the experiment start date. For example, if someone received a "day before appointment" notification for an appointment on the first day of the experiment and came in same day. Frankly, I'd like to refine the experiment's concept of start/end dates. It would be better to have `experiment.notification_start_date` and `experiment.appointment_start_date` or something like that, given the difficulty we experienced even talking about this.